src/ODE.jl

```diff
@@ -251,6 +251,17 @@ function ode23s(F, y0, tspan; reltol = 1.0e-5, abstol = 1.0e-8,
 
     # initialization
     t = tspan[1]
+    
+    # Component-wise Tolerance check similar to MATLAB ode23s
+    # Support for component-wise absolute tolerance only.
+    # Relative tolerance should be a scalar.
+    @assert length(reltol) == 1 "Relative tolerance must be a scalar"
```

Comment:
This would only apply to reltol in the case of ode23s. For the other tolerances it would not be possible since they can be either a scalar or a vector of the same length as the problem.